config.def.h

```diff
@@ -178,4 +178,12 @@ static const cursor_t imgcursor[3] = {
 	CURSOR_LEFT, CURSOR_ARROW, CURSOR_RIGHT
 };
 
+/* width of navigation area on left and right side of the window:
+ * a negative value defines the width as number of pixels (without the negative sign),
+ * a positive value defines the width as percentage of the window width,
+ * 0 disables cursor navigation,
+ * the area width will be kept under half of the window width in any case.
+ */
+static const int NAV_WIDTH = 33;
+
```

Comment:
This seems way to convoluted. Why not just a float that defines the percentage?

Reply:
Oh, that would be my initial commit :) I found the current way is cleaner.

Reply:
I think it should just be a float, and anything above 1 or below 0 should just be ignored. Maybe even anything above 0.5, because 0.5 means you've turned the entire window into navigation.

Reply:
I commented about that up here: https://github.com/nsxiv/nsxiv/pull/155#discussion_r741574409

Reply:
The current way of doing this is just confusing, evident by the fact that we need a long multi-line comment to describe it. If we do want absolute width then we should have a switch for it.

```c
static const bool is_abs_nav = false;
```

Also I don't think we "need" to have absolute width. For now let's just keep it to be percentage. If someone comes along and thinks having absolute width is a killer feature then we can maybe re-consider.

Reply:
The absolute width can also be a community patch in https://github.com/nsxiv/nsxiv-extra now that I think about it.

Reply:
That "someone" would include me. I think absolute width has its own advantage, since the width will be the same no matter what the window size is, thus the user will always know exactly how far from the left/right border is the navigation area. He can get familiar with that size. Different users might prefer different measurement methods, relative or absolute, depending on their thinking habits.

As for the long comment, I think if well worded, it's actually easy to understand.

If you are OK either with or without it, I would ask to leave it be, but if you think that's an absolutely bad thing, then I can make it a single float for relative width.

Reply:
I will prefer not adding absolute width, but no strong opinion.

If you want to add it then please make it a bool switch. That way we can have this.

```c
static const bool is_abs_nav = false;
static const unsigned int nav-width = 33;
```

would be 33% . And someone uses `is_abs_nav` true then it will be 33 pixles.

The names of the vars can be improved. But I think you get the point. It would also make the chnages a lot more readable, e.g `if (abs_nav)` compared to what we're currently doing by checking negavite value.